### ISSUE TITLE
Fails to load and save the personal links

### ISSUE THREAD

**Issue by shb2002** [Adam-it/TilesLinksForSPOnline#21]
I deployed the app to a SPO site and added the web part to a page. The loading icon stays on the page all the time. Added a few personal links, dragged them around with no issue, but it doesn't save my links so they are all lost upon a refresh.

Looking at the browser console I see below error occurs on all operation such as load, add a new link, dragging them around, etc.

![image](https://user-images.githubusercontent.com/77365068/143514553-2726977b-88c6-471e-bb33-05d54624f5c1.png)

It seems it can't find the links json in the repository which make sense for the first time using this component on a fresh tenant, however it never creates one. It seems it's trying to load the data first on every operation, but then it stop upon failure to load and never create one.

Would you please help with the issue? Is there any additional setting needs to be done at MS Graph or Azure AD side?

**Comment by Adam-it** [comment]
hi,
did you check and approve all api permissions request from SharePoint admin site (MSD documentation - deploy the solution and grant permissions)
You may check if there are any pending permissions in Api access page (if yes please approve them)
![image](https://user-images.githubusercontent.com/58668583/143597797-f73ea9b1-e09d-4964-912b-399d4bd985c5.png)
also on the user on which you testes the webpart could you check if in the user OneDrive in MyFiles/applications you see a 
SharePoint Online Client Extensibility Web Application Principal and a PersonalTipesApp folder in it 🤔? 
![image](https://user-images.githubusercontent.com/58668583/143598024-d3c642d4-139f-4671-90d4-cf1986167629.png)

**Comment by shb2002** [comment]
Thanks Adam for the reply.
I followed with the API Access page and approved the permission request. Then it started working for the user deployed the web part.
Then I signed in as a different user and browsed to the page with the web part. I got the same issue as last time. The loading icon was staying on all the time and the added tiles can be saved. This time the error message is slightly different:

![image](https://user-images.githubusercontent.com/77365068/144179424-e82f8e45-6403-4d71-8883-bd66f9d1737e.png)

I checked the API Access page and there is no pending access request. No file in OneDrive either.

**Comment by Adam-it** [comment]
@shb2002  thanks for reaching out. Ok I think I understand the issue. TBH it is a bit strange and it should be working fine also for the other user as well. I will try testing this scenario one more time on a clean tenant and will let you know if I find anything. 
Thanks for reporting this, now it's m turn to start the investigation what might be wrong if I reproduce it 👍

**Comment by shb2002** [comment]
Thanks Adam! 
Just an update on this issue. It's now started working for other users as well without doing any change. It seems it takes some time (over a day) for the privileges to go through.

Please feel free to close this issue.

**Comment by Adam-it** [comment]
@shb2002 aaa now way, I already started working on this and got so engaged ... only joking, I just started with coffee like always before coding 😅.
Anyway thanks for the comment and info. It's great news it started working 👍👍👍. TBH I am not sure if it takes some time for the privileges to start working 🤔. I think it should start working instantly but then again I might be wrong 🤔.
Ok. Closing the issue. I hope you will have some fun with the solution and feel free if you would like to submit any idea or a PR 👍
